modules/core/src/main/scala/com/azavea/stac4s/ItemCollection.scala

```diff
@@ -1,42 +1,75 @@
 package com.azavea.stac4s
 
 import cats.Eq
+import cats.implicits._
 import io.circe._
 import io.circe.refined._
+import io.circe.syntax._
 
 final case class ItemCollection(
     _type: String = "FeatureCollection",
     stacVersion: StacVersion,
     stacExtensions: List[String],
     features: List[StacItem],
-    links: List[StacLink]
+    links: List[StacLink],
+    extensionFields: JsonObject
 )
 
 object ItemCollection {
 
-  implicit val eqItemCollection: Eq[ItemCollection] = Eq.fromUniversalEquals
-
-  implicit val encItemCollection: Encoder[ItemCollection] = Encoder.forProduct5(
+  val itemCollectionFields = Set(
     "type",
     "stac_version",
     "stac_extensions",
     "features",
     "links"
-  )(itemCollection =>
-    (
-      itemCollection._type,
-      itemCollection.stacVersion,
-      itemCollection.stacExtensions,
-      itemCollection.features,
-      itemCollection.links
-    )
   )
 
-  implicit val decItemCollection: Decoder[ItemCollection] = Decoder.forProduct5(
-    "type",
-    "stac_version",
-    "stac_extensions",
-    "features",
-    "links"
-  )(ItemCollection.apply)
+  implicit val eqItemCollection: Eq[ItemCollection] = Eq.fromUniversalEquals
+
+  implicit val encItemCollection: Encoder[ItemCollection] = new Encoder[ItemCollection] {
+
+    def apply(collection: ItemCollection): Json = {
+      val baseEncoder: Encoder[ItemCollection] = Encoder.forProduct5(
+        "type",
+        "stac_version",
+        "stac_extensions",
+        "features",
+        "links"
+      )(itemCollection =>
+        (
+          itemCollection._type,
+          itemCollection.stacVersion,
+          itemCollection.stacExtensions,
+          itemCollection.features,
+          itemCollection.links
+        )
+      )
+
+      baseEncoder(collection).deepMerge(collection.extensionFields.asJson)
+    }
+  }
+
+  implicit val decItemCollection: Decoder[ItemCollection] = { c: HCursor =>
+    (
+      c.get[String]("type"),
+      c.get[StacVersion]("stac_version"),
+      c.get[Option[List[String]]]("stac_extensions"),
+      c.get[List[StacItem]]("features"),
+      c.get[Option[List[StacLink]]]("links"),
+      c.value.as[JsonObject]
+    ).mapN(
+      (
+          _type: String,
+          stacVersion: StacVersion,
+          extensions: Option[List[String]],
+          features: List[StacItem],
+          links: Option[List[StacLink]],
+          document: JsonObject
+      ) =>
+        ItemCollection(_type, stacVersion, extensions getOrElse Nil, features, links getOrElse Nil, document.filter({
+          case (k, _) => !itemCollectionFields.contains(k)
```

Comment:
Ah ok, I think I see what's happening here. The whole document gets read and anything not in the defined fields gets lopped into the `extensionFields`. I was originally confused because I thought maybe `document` should be `extensionFields` above but I think this makes sense.

Reply:
It's not hard to ensure that no fields from the case class are also keys in the `extensionFields` json object -- we can use [`LabelledGeneric`](https://svejcar.dev/posts/2019/10/22/extracting-case-class-field-names-with-shapeless/) from shapeless to extract the field names, and I think we can just map `_type` => `type` in the names and end up with something that mostly works. I'm trying to think of whether this is a terrible idea for some reason I haven't thought of yet. I have end-of-day brain though so I'll think about it again tomorrow morning.

I actually ran into this issue while working on this PR -- I'd forgotten the `filter` in collection and had about a half hour of trouble figuring out what was wrong with my collection json round trips.